paper/paper.md

```diff
@@ -0,0 +1,167 @@
+University of Pretoria, University of Edinburgh, Northeastern University,
+Central Connecticut State University, KTH Royal Institute of Technology, CU
+Boulder, University of Amsterdam, George Washington University, Texas A&M, Bryn
+Mawr, and University of Maryland; and, as of May 2018, over 10,000
+nbgrader-based notebooks exist on GitHub. In addition to its core functionality,
+nbgrader has expanded to support a number of other features, including the
+ability to handle multiple courses on the same JupyterHub instance; the option
+to either include or hide autograder tests; customizable late penalties; and
+support for importing assignment files downloaded from a Learning Management
+System (LMS).
+
+# Statement of Need
+
+The use of computational methods has become increasingly widespread in fields
+outside of computer science [@wing2008computational]. As these disciplines begin
+to require computational tools, undergraduate curricula also begin to include
+topics in programming and computer science. However, because students are
+focused on the discipline that is the object of their study—and programming is
+likely a secondary interest—teaching students through traditional computer
+science offerings is not always effective [@Cortina2007; @Forte2005;
+@Guzdial2005]. While there are visual programming languages such as Raptor
+[@Carlisle2004] and Scratch [@Resnick2009] that are intended to be easy and
+enjoyable for those unfamiliar with programming to learn, they lack specialized
+tools such as numerical or data visualization libraries, which are required for
+effective work in domain sciences. A hybrid approach is to teach students
+computational concepts in an interactive environment where it is possible to
+quickly write, test, and tweak small units of code. Many such environments
+exist, including Mathematica [@mathematica], Maple [@maple], MATLAB [@matlab],
+Sage [@sage] and IPython [@PerezGranger2007].
+
+In recent years, the IPython project introduced the *Jupyter notebook*
+[@kluyver2016jupyter], an interface that is conducive to interactive and
+literate computing, where programmers can interleave prose with code and
+figures. The Jupyter notebook is ideal for educators because it allows them to
+create assignments which include instructions along with code or Markdown cells,
+in which students can provide solutions to exercises. Students can, for example,
+write code both to compute and visualize a particular result. Because the
+notebook is interactive, students can iterate on a coding problem without
+needing to switch back and forth between a command line and a text editor, and
+they can rapidly see results alongside the code which produced them.
+
+Instructors in many fields have begun using the Jupyter notebook as a teaching
+platform. The notebook has appeared in over 70 classes
+[@Castano_Jupyter_Map_Dataset] on subjects including geology, mathematics,
```

Comment:
Also, remember the spreadsheet survey that was done earlier. I think it has to be hundreds of classes now, right?

Reply:
The Jupyter Map includes all the courses from that survey, I believe.

Reply:
I should say, I am sure there are hundreds of courses using Jupyter. I just don't have a source to back that up unfortunately 😞

Reply:
I don't think the map includes all of those. There are a few more at Bryn Mawr College that are not on the map, because those instructors did not respond. If you just take the number of rows in the survey, I think it is much larger, no?

Reply:
Perhaps the 10k nbgrader notebooks could b used to help approximate the number of courses?

Reply:
You're right, there are 125 rows in that survey: https://github.com/jupyter/surveys/blob/master/surveys/2016-05-education-survey/responses.csv

There are surely duplicates between the map and the survey though; meh. 

> Perhaps the 10k nbgrader notebooks could b used to help approximate the number of courses?

Probably; I unfortunately don't think I will have time to write a script to do that analysis though---and there are surely many duplicates, and notebooks that someone just tried out which never became part of a class, etc. Maybe someone else can volunteer? 😄